app/models/orchestration.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "velum/salt"
+require "velum/salt_orchestration"
+
+# Orchestration represents a salt orchestration event
+class Orchestration < ApplicationRecord
+  class OrchestrationAlreadyRan < StandardError; end
+
+  enum kind: [:bootstrap, :upgrade]
+  enum status: [:in_progress, :succeeded, :failed]
+
+  after_create :update_minions
+
+  # rubocop:disable Rails/SkipsModelValidations
+  def run
+    raise OrchestrationAlreadyRan unless jid.blank?
+    update_column :status, Orchestration.statuses[:in_progress]
+    _, job = case kind
+             when "bootstrap"
+               Velum::Salt.orchestrate
+             when "upgrade"
+               Velum::Salt.update_orchestration
+    end
+    update_column :jid, job["return"].first["jid"]
```

Comment:
Why don't you use `update_columns` and set both `status` and `jid` in the same statement ?

Reply:
Because you need to set `status` to `in_progress`, then call the orchestration, and then save the `jid` that was returned by calling the orchestration.

Reply:
Gotcha